contracts/lease/src/lease/with_lease_deps.rs

```diff
@@ -72,7 +72,7 @@ where
 
     fn on<C>(self) -> AnyVisitorResult<Self>
     where
-        C: 'static + Currency + Serialize + DeserializeOwned,
+        C: 'static + Currency + DeserializeOwned,
```

Comment:
I'm curious to see if there are `Deserialize` and `DeserializeOwner` extra trait boundaries.